### DISCUSSION TITLE
Talos on Raspberry PI CM4 Lite models using PXE netboot

### DISCUSSION THREAD

**Discussion by chrsoo** [siderolabs/talos#4769] [Q&A]
Hi,

Having spent a few hours browsing through the Talos documentation I am still unclear on the hardware requirements of running Talos on Raspberry PI CM 4 *Lite* modules - can this be done without an SD card or other local persistent storage?

My goal is a **diskless** K8s cluster with worker nodes booting using netboot - only the master node will have local storage. Planning to host this on the [Turing PI 2](https://turingpi.com/) once it is available.

Any input is most appreciated!

**Comment by smira** [comment]
Talos doesn't quite support diskless mode. It can be PXE booted over the network, but it still requires some disk where `/var` partition will be created.

**Reply by chrsoo** [reply]
Thanks for the reply!

Can the /var partition be created on a ramdisk or on an nbd/nfs share?

**Reply by smira** [reply]
It's not supported at the moment. Technically it should be possible, but might require significant changes to the code.

**Reply by BryceAshey** [reply]
Working on PXE booting a 24-node BitScope cluster (RPi4).

just adding my +1 to this - I ran into the same thing - PXE boot works great all the way to the point where it tries to mount the disk. With no way to supply something other than a physically attached drive limits the use cases a little. For now we'll leave the SD cards in - it will be interesting to see what sort of lifetime we get out of them.

**Reply by Arthurdw** [reply]
Hi, quick question if this is still the case and if it is if it us possible to use a network drive for this /var usage?

Thanks in advance for the reply :)
